ropetest/pyscopestest.py

```diff
@@ -330,3 +342,56 @@ def test_getting_defined_names_for_modules(self):
         self.assertTrue("A" in scope.get_names())
         self.assertTrue("open" not in scope.get_defined_names())
         self.assertTrue("A" in scope.get_defined_names())
+
+    def test_get_inner_scope_for_list_comprhension_with_many_targets(self):
+        scope = libutils.get_string_scope(
+            self.project, "a = [(i, j) for i,j in enumerate(range(10))]\n"
+        )
+        self.assertEqual(len(scope.get_scopes()), 1)
+        self.assertNotIn("i", scope)
+        self.assertNotIn("j", scope)
+        self.assertIn("i", scope.get_scopes()[0])
+        self.assertIn("j", scope.get_scopes()[0])
+
+    def test_get_inner_scope_for_generator(self):
+        scope = libutils.get_string_scope(self.project, "a = (i for i in range(10))\n")
+        self.assertEqual(len(scope.get_scopes()), 1)
+        self.assertNotIn("i", scope)
+        self.assertIn("i", scope.get_scopes()[0])
+
+    def test_get_inner_scope_for_set_comprehension(self):
+        scope = libutils.get_string_scope(self.project, "a = {i for i in range(10)}\n")
+        self.assertEqual(len(scope.get_scopes()), 1)
+        self.assertNotIn("i", scope)
+        self.assertIn("i", scope.get_scopes()[0])
+
+    def test_get_inner_scope_for_dict_comprehension(self):
+        scope = libutils.get_string_scope(
+            self.project, "a = {i:i for i in range(10)}\n"
+        )
+        self.assertEqual(len(scope.get_scopes()), 1)
+        self.assertNotIn("i", scope)
+        self.assertIn("i", scope.get_scopes()[0])
+
+    def test_get_inner_scope_for_nested_list_comprhension(self):
+        scope = libutils.get_string_scope(
+            self.project, "a = [[i + j for j in range(10)] for i in range(10)]\n"
+        )
+
+        self.assertEqual(len(scope.get_scopes()), 1)
+        self.assertNotIn("i", scope)
+        self.assertNotIn("j", scope)
+        self.assertIn("i", scope.get_scopes()[0])
+        self.assertEqual(len(scope.get_scopes()[0].get_scopes()), 1)
+        self.assertIn("j", scope.get_scopes()[0].get_scopes()[0])
+        self.assertIn("i", scope.get_scopes()[0].get_scopes()[0])
+
+
+def suite():
+    result = unittest.TestSuite()
+    result.addTests(unittest.makeSuite(PyCoreScopesTest))
+    return result
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
As we are using `pytest` now, it is no longer necessary to create these Test Suites and main() constructs.